Back out "Back out "[aotinductor] Add example_value metadata to nodes (#112415)""

frank-wei opened the pull request:
Reviewed By: khabinov

Differential Revision: D51223990




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

zhxchen17 reviewed (changes requested):
I'm not sure why we need to run fake prop here. Could you explain?

frank-wei commented:
> I'm not sure why we need to run fake prop here. Could you explain?

Some of the following passes need to use certain tensor's meta["example_value"].shape in their condition check.

zhxchen17 reviewed (commented):
Yeah the contract makes sense but I still have some question with the code placement.

Commit:
Back out "Back out "[aotinductor] Add example_value metadata to nodes (#112415)"" (#113607)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/113607

Reviewed By: khabinov, angelayi

Differential Revision: D51223990

frank-wei commented on torch/_export/__init__.py:
```diff
@@ -955,6 +955,10 @@ def aot_compile(
     gm = _export_to_torch_ir(f, args, kwargs, constraints)
+
+    from torch._export.passes.fake_tensor_prop import FakeTensorProp
+    FakeTensorProp(gm).run()
```
Moved it.

zhxchen17 reviewed (approved):
thanks for addressing my questions!
